maintenance/projects/js-toolkit/packages/liferay-npm-build-support/src/loader/normalize-angular-output.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+import {
+	BundlerLoaderContext,
+	BundlerLoaderReturn,
+} from 'liferay-npm-build-tools-common/lib/api/loaders';
+
+/**
+ * This loader:
+ *
+ *   1. Discards any webpack artifact not containing ES5 code
+ *   2. Removes `-es5` from file and module names when present
+ *
+ * This way, the bundler can treat any known Angular version the same, no matter
+ * how webpack produced its output.
+ *
+ * This is because Angular version <=10 produce artifacts for ES5 and ES2015
+ * targets, while Angular versions >=10 only produce one type of artifact, not
+ * differentiated by file name.
+ */
+export default function (context: BundlerLoaderContext): BundlerLoaderReturn {
+	const {filePath, log} = context;
+
+	if (filePath.includes('-es') && !filePath.includes('-es5.')) {
+		context.content = undefined;
+		context.extraArtifacts = {};
+
+		log.info(
+			'normalize-angular-output',
+			`Discarding output because source is not ES5 code`
+		);
+
+		return undefined;
```

Comment:
This line offends me.

![offended](https://user-images.githubusercontent.com/7074/110771811-37848a80-825b-11eb-8bf1-33c2a4757d9b.gif)